### PR TITLE
dbd: Unify and rewrite recipe

### PR DESCRIPTION
**Pull request by eric-ch**
dbd and db-tools are both built from the same repository and from the
same path in the directory. The OCaml toolchain does not separate source
and build artifacts. This leads to a world of troubles when using
externalsrc since both rely on the local findlib database.

There is little reason dbd and db-tools are built by separate recipes.
Unify all that and take the opportunity to fix the following:
- db-cat is overwritten by db-write due to a typo in the recipe
- remove haskell.bbclass, nothing uses it
- Use B where appropriate, but...
- brokensep (B = ${S}) is still required as the Makefile references
  autogen directly
- Fix RDEPENDS on libargo

### CODE REVIEW

**Comment by eric-ch** [comment]
Build [$10048](http://openxt-builder.ainfosec.com:8010/#/builders/3/builds/10048)

File: recipes-openxt/dbd/dbd_git.bb [hunk 1]
```diff
@@ -9,48 +10,103 @@ DEPENDS = " \
...
+    ${bindir}/db-cmd \
+    ${bindir}/db-exists \
+    ${bindir}/db-ls \
+    ${bindir}/db-nodes \
+    ${bindir}/db-read \
+    ${bindir}/db-rm \
+    ${bindir}/db-write \
+    ${bindir}/db-cat \
+"
+RDEPENDS_${PN}-tools += "bash"
```
Comment (jandryuk) [inline recipes-openxt/dbd/dbd_git.bb]: db-$foo and db-$foo-dom0 are both `#!/bin/sh`, so I don't think there is a bash RDEPENDS for `${PN}-tools`.  dbd.initscript is `#!/bin/bash` with bashisms though.

Other than that, LGTM.

**Comment by eric-ch** [comment]
#### v2:
- Removed `RDEPENDS` on `bash` for `dbd-tools`

**Comment by jandryuk** [comment]
Will merge soon

**Pull request merged**